### PR TITLE
Update UI

### LINKED ISSUE #2: Update UI
### Description

Update UI to adopt it to Lisk network

### Acceptance Criteria

- Update UI 
- Users should be able to claim LSK tokens with UI

### PR DESCRIPTION
**Pull request by nagdahimanshu**
### What was the problem?

This PR resolves #2 #4 

### How was it solved?

- [x] Update UI
- [x] Update readme to make generic ERC20 faucet
- [x] Add Makefile
- [x] Enable CI
- [x]  Fix existing unit test
- [x] Add unit tests
  - [x] LSKtoWei
  - [x] TransferERC20
  - [x] addLeftPadding

### How was it tested?
Local + CI

### CODE REVIEW

**Review by hrmhatef — CHANGES_REQUESTED** [review]
LGTM, just small suggestion: `skip-dirs` is deprecated and we should use `exclude-dirs` based in the documents of the `golangci-lint` [here](https://golangci-lint.run/usage/configuration/#issues-configuration)

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+BLUE = \033[1;34m
+GREEN = \033[1;32m
+COLOR_END = \033[0;39m
+
+build: build-frontend build-backend
+
+build-backend: # Builds the application and create a binary at ./bin/
+	@echo "$(BLUE)» Building $(APP_NAME) application binary... $(COLOR_END)"
+	@go build -a -o bin/$(APP_NAME) .
+	@echo "$(GREEN) Binary successfully built$(COLOR_END)"
```
Comment (sameersubudhi) [inline Makefile]:
```suggestion
	@echo "$(GREEN)Binary successfully built$(COLOR_END)"
```

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+build-frontend: # Builds the frontned application
+	@echo "$(BLUE)» Building frontend... $(COLOR_END)"
+	@go generate
+	@echo "$(GREEN) Frontend successfully built$(COLOR_END)"
```
Comment (sameersubudhi) [inline Makefile]:
```suggestion
	@echo "$(GREEN)Frontend successfully built$(COLOR_END)"
```

File: README.md [hunk 3]
```diff
@@ -7,7 +7,7 @@
 ![GitHub issues](https://img.shields.io/github/issues-raw/liskhq/lsk-faucet)
 ![GitHub closed issues](https://img.shields.io/github/issues-closed-raw/liskhq/lsk-faucet)
 
-LSK faucet is a web application to get Lisk (LSK) tokens on the Lisk Sepolia Testnet. The tokens can be used to test and troubleshoot your decentralized application or protocol before going live on the Lisk Mainnet.
+LSK faucet is a web application that can be configured and deployed to get custom ERC20 tokens on any network. The tokens can be used to test and troubleshoot your decentralized application or protocol before going live on the Mainnet.
```
Comment (sameersubudhi) [inline README.md]:
```suggestion
LSK faucet is a web application that can be configured and deployed to get ETH and custom ERC20 tokens on any test network. The tokens can be used to test and troubleshoot your decentralized application or protocol before going live on the Mainnet.
```

File: README.md [hunk 4]
```diff
@@ -32,38 +32,41 @@ cd lsk-faucet
 ```
 
 2. Bundle Frontend web with Vite
+
+**NOTE**: Please make sure to replace token icon at `web/public/token.png` with the specific ERC20 token icon.
```
Comment (sameersubudhi) [inline README.md]: Would it make sense to replace the note with something like below? The following is just a format recommendation where we must list all the supported image formats and recommend the image resolution.

```suggestion
**NOTE**: Please make sure to update the token icon under `web/public/` with the specific ERC20 token icon. The file must be named either `token.png` or `token.jpg`. We recommend the image dimensions to be 50px x 50px.
```

File: README.md [hunk 5]
```diff
@@ -32,38 +32,41 @@ cd lsk-faucet
...
 ```bash
-./lsk-faucet -httpport 8080 -wallet.provider http://localhost:8545 -wallet.keyjson keystore -wallet.keypass password.txt
+make run FLAGS="-httpport 8080 -wallet.provider http://localhost:8545 -wallet.keyjson keystore -wallet.keypass password.txt"
 ```
 
 ### Configuration
 Below is a list of environment variables that can be configured.
 
-- `WEB3_PROVIDER`: Endpoint for Lisk JSON-RPC connection.
+- `WEB3_PROVIDER`: RPC Endpoint to connect with the network.
```
Comment (sameersubudhi) [inline README.md]:
```suggestion
- `WEB3_PROVIDER`: RPC Endpoint to connect with the network node.
```

File: README.md [hunk 6]
```diff
@@ -32,38 +32,41 @@ cd lsk-faucet
...
 - `PRIVATE_KEY`: Private key hex to fund user requests with.
 - `KEYSTORE`: Keystore file to fund user requests with.
 - `HCAPTCHA_SITEKEY`: hCaptcha sitekey.
 - `HCAPTCHA_SECRET`: hCaptcha secret.
-- `LSK_TOKEN_ADDRESS`: Contract address of LSK token on the Lisk L2.
+- `ERC20_TOKEN_ADDRESS`: Contract address of ERC20 token on the above mentioned network, defaults to Lisk L2 token address.
```
Comment (sameersubudhi) [inline README.md]: What do we mean by `above-mentioned network`? I do not see any specified networks. Can we please rephrase it to be more clear?
```suggestion
- `ERC20_TOKEN_ADDRESS`: Contract address of the ERC20 token on the above mentioned network, defaults to contract address for Lisk ERC20 tokens on Lisk Sepolia.
```

File: README.md [hunk 7]
```diff
@@ -79,46 +82,45 @@ echo "your keystore password" > `pwd`/password.txt
...
-| -faucet.symbol    | Token symbol to display on the frontend          | LSK                                        |
-| -hcaptcha.sitekey | hCaptcha sitekey                                 |                                            |
-| -hcaptcha.secret  | hCaptcha secret                                  |                                            |
+| Flag              | Description                                         | Default Value                              |
+| ----------------- | --------------------------------------------------- | ------------------------------------------ |
+| -httpport         | Listener port to serve HTTP connection              | 8080                                       |
+| -proxycount       | Count of reverse proxies in front of the server     | 0                                          |
+| -token-address    | Token contract address                              | 0x8a21CF9Ba08Ae709D64Cb25AfAA951183EC9FF6D |
+| -faucet.amount    | Number of ERC20 tokens to transfer per user request | 1                                          |
```
Comment (sameersubudhi) [inline README.md:97]: Is it safe to assume that all ERC20 tokens can be divided into 10^18 parts (similar to 1 ETH == 10^18 Wei)?
If not, should we also add the support for `decimals`?

File: internal/chain/util_test.go [hunk 8]
```diff
@@ -48,3 +50,41 @@ func TestEtherToWei(t *testing.T) {
...
 	}
 }
+
+func Test_LSKToWei(t *testing.T) {
+	tests := []struct {
+		name   string
+		amount int64
+		want   *big.Int
+	}{
+		{name: "Should convert 1 LSK to Wei", amount: 1, want: big.NewInt(1000000000000000000)},
```
Comment (sameersubudhi) [inline internal/chain/util_test.go]:
```suggestion
		{name: "should convert 1 LSK to 10^18 Wei", amount: 1, want: big.NewInt(1000000000000000000)},
```

File: internal/chain/util_test.go [hunk 9]
```diff
@@ -48,3 +50,41 @@ func TestEtherToWei(t *testing.T) {
...
+}
+
+func Test_addLeftPadding(t *testing.T) {
+	tests := []struct {
+		name  string
+		input []byte
+		want  []byte
+	}{
+		{
+			name:  "Should return padded input",
```
Comment (sameersubudhi) [inline internal/chain/util_test.go]:
```suggestion
			name:  "should return padded input",
```

File: web/src/Faucet.svelte [hunk 10]
```diff
@@ -189,8 +189,8 @@
 <style>
   .hero.is-info {
     background:
-      linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)),
-      url('/background.jpg') no-repeat center center fixed;
+      linear-gradient(rgba(0, 0, 0.5, 0), rgba(0, 0, 0.5, 0)),
```
Comment (sameersubudhi) [inline web/src/Faucet.svelte]: How did we decide on the rgba values?

**Review by sameersubudhi — CHANGES_REQUESTED** [review]
Looks good overall. But could you please clarify/look at the following doubts/recommendations? 🙏

File: README.md (hunk 7 above)
Comment (nagdahimanshu) [reply README.md:97]: Discussed offline.

File: web/src/Faucet.svelte (hunk 10 above)
Comment (nagdahimanshu) [reply web/src/Faucet.svelte]: Updated as per portal.

**Review by sameersubudhi — APPROVED** [review]
The changes LGTM 👍 
As discussed, let's create a separate PR for the unit test for `TransferERC20` method.

**Review by hrmhatef — APPROVED** [review]
LGTM👍

**Review by sameersubudhi — APPROVED** [review]

**Review by hrmhatef — APPROVED** [review]
